secure_messaging.c

```diff
@@ -102,14 +102,14 @@ void secure_messaging_increment_context(SecureMessaging* secure_messaging) {
     } while(++context[--context_len] == 0 && context_len > 0);
 }
 
-void secure_messaging_wrap_apdu(
-    SecureMessaging* secure_messaging,
-    uint8_t* message,
-    size_t message_len,
-    BitBuffer* tx_buffer) {
+void secure_messaging_wrap_apdu(SecureMessaging* secure_messaging, BitBuffer* tx_buffer) {
```

Comment:
I can't recall exactly why I added the extra params: maybe I was concerned that I would be mutating the BitBuffer and could lose data?  This would be another thing to have in it's own PR so I could test it on it's own.

Reply:
It's actually a change that I've made and will split out in a PR. I have adapted passy to use the Tx buffer to compile data to be send. If it has to be encrypted, because secure messaging has started, it will encrypt the message and put it back in the Tx buffer.

This allows for simpler functions that can be used with or without SM.